### PR TITLE
[enhancement](Nereids) add type coercion between decimal and integral

### PR DESCRIPTION
**Pull request by Kikyou1997**
# Proposed changes

Issue Number: noissue
## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
@morrySnow

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java [hunk 1]
```diff
@@ -189,6 +192,8 @@ public static Optional<DataType> findTightestCommonType(DataType left, DataType
             }
         } else if (left instanceof CharacterType || right instanceof CharacterType) {
             tightestCommonType = StringType.INSTANCE;
+        } else if (left instanceof DecimalType && right instanceof IntegralType) {
+            tightestCommonType = new DecimalType(9, 0);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java:196]:
```suggestion
            tightestCommonType = DecimalType.forType(right);
```

**Review by morrySnow — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java [hunk 2]
```diff
@@ -189,6 +193,10 @@ public static Optional<DataType> findTightestCommonType(DataType left, DataType
             }
         } else if (left instanceof CharacterType || right instanceof CharacterType) {
             tightestCommonType = StringType.INSTANCE;
+        } else if (left instanceof DecimalType && right instanceof IntegralType) {
+            tightestCommonType = DecimalType.forType(right);
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java:197]: What about this case:
 - left: Decimal(20,0)
 - right: int
Here the tightestCommonType is Decimal(10, 0), which cannot cover left.
Is this a problem?

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java:197]: yes, should use 
```
DecimalType.widerDecimalType(left, DecimalType.forType(right));
```

Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java:197]: it would be better to add this case as a UT

Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java:197]: you are right, thx pro

Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java:197]: you are right, thx bro

**Review by englefly — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**